### ISSUE TITLE
Bones rotated incorrectly

### ISSUE THREAD

**Issue by kai13xd** [niacdoial/blemd#30]
Hey this is Kai from the Pikmin hacking community. We are currently starting to get into custom animations and I've noticed that this plugin gives the best results when it comes to weighting the mesh and placing bones in the correct places for Blender users. When importing a bmd file from Pikmin 2, the bones seem to be rotated incorrectly and need to be moved 90 degrees in the Z axis. Also is there a way to make intermediary bones to be made into a bone chains by connecting them head to tail? That would make it easier for recreating IK systems without having to readjust each bone manually to neaten them.
![3af510d97625e0c3f4419f347b26b6e8](https://user-images.githubusercontent.com/7115703/60381433-8902cc80-9a1a-11e9-926a-ccb81d2a4984.png)

**Comment by niacdoial** [comment]
[TL;DR version below]
Sadly, it would be very hard to do *with the animations*, due to several things.
- In BMD format, the bones are only defined by their head and orientation of said head, meaning that the bones' *true* default positions do not take in account limb orientation at all.
- In blender, editing the position of a bone's tail completely changes the way the pose orientation is defined. (but you probably knew this one lol)
- In the BMD file, the armature *true* default pose corresponds to a very crumpled mesh, and the actual rest pose is defined as with the armature's *effective* default pose. (this means that an animation can play faster by simply replacing the effective default pose instead of being intertwined with it)
- The way BleMD works is by applying the effective default pose to the mesh to put the bones in a reasonable position
- BMD uses `+Z -> front` and `+Y -> up` while blender+BleMD uses `+Z -> up` (and `+Y -> back`).

All of this means that the animations have to be converted, and the mathematical process is already complicated, that already took me probably more than ten hours to get right (and formatted in a somewhat human-readable way).
What can be done more easily, though, is to discard the animations altogether in order to do this. But this would give the same result as running a custom script to align the bones after the fact.

However, if you want to try to understand and adjust the math to take the animations into account while doing what you asked, you can. Just be sure to yell at me for the parts that aren't commented well/enough

##  TL;DR:
There is a way to do what you asked, but either by discarding animations, or editing the already complex and headache-inducing math behind the way animations are treated.

**Comment by niacdoial** [comment]
(I implemented this, under the `useatural bone placement` option)
